src/App.tsx

```diff
@@ -17,18 +77,39 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                query={query}
+                onFilterChange={handleFilterChange}
+                onSearchChange={handleSearchChange}
+                onClearSearch={handleClearSearch}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading ? (
+                <Loader />
```

Comment:
Consider using the `classnames` library for conditional class management as specified in the task requirements. This will enhance code readability and maintainability.